examples/mugen/retrieval/s3d.py

```diff
@@ -30,17 +30,17 @@ def __init__(self, num_class):
             BasicConv3d(64, 64, kernel_size=1, stride=1),
```

Comment:
I think we can make a more generic class that able to handle both `S3D` and `I3D` (see https://arxiv.org/pdf/1705.07750.pdf). The idea (just to illustrate, the naming might not be good) is to enable user to choose the `class` for the `conv_block (S3D: SepConv3D, I3D: Conv3D)`, `max_pool_block`, and `inception_block (S3D: SepInceptionModule3D, I3D: InceptionModule3D)` (examples: [here](https://github.com/pytorch/vision/blob/main/torchvision/models/video/mvit.py#L352) or [here](https://github.com/pytorch/vision/blob/main/torchvision/models/efficientnet.py#L306)). 

It is also good to let the parameters also be configurable on the generic class, this will be useful if there will be a new variant with different parameters in the future. This is not a must, however since there are a lot of parameters, it might be a good idea to create a config class to store the parameters ([example](https://github.com/pytorch/vision/blob/main/torchvision/models/efficientnet.py#L377)).

Once we have the generic module, we can have a builder to build the s3d model ([example](https://github.com/pytorch/vision/blob/main/torchvision/models/video/mvit.py#L464))